routes/songlist.js

```diff
@@ -26,42 +66,46 @@ router.get('/', function(req, res) {
 
 router.post('/', function(req, res) {
     var id = req.body.id
-    console.log(typeof(id))
     if (req.isAuthenticated()) {
         User.findOne({
-            username: req.user.username
+            _id: req.user._id
         }, function(err, user) {
             if (err) return handleError(err)
             User.findOne({
-                    library: id
-                })
-                .populate('library')
-                .exec(function(err, song) {
-                    console.log(JSON.stringify(song))
-                    if (err) return handleError(err)
-                    if (song) {
-                        console.log('hahhaa')
+                library: id
+            }, function(err, song) {
+                if (err) return handleError(err)
+                // console.log(song)
+                if (song) {
```

Comment:
What is this doing? The if versus the else? One is deleting and the other is adding?